libpromises/generic_agent.c

```diff
@@ -831,7 +845,7 @@ void LoadAugments(EvalContext *ctx, GenericAgentConfig *config)
     else
     {
         def_json = StringFormat("%s%c%s", config->input_dir, FILE_SEPARATOR, "def_preferred.json");
-        if (access(def_json, R_OK) != 0)
+        if (!IsFile(def_json))
```

Comment:
Shouldn't this still check if the file is readable?

Reply:
@vpodzime Should it? Actually I am not sure. This is only to decide between `def.json` and `def_preferred.json`. After that, the behavior is the same as before. So if it is not readable, it would later fail to open the file for reading, and give some error. I guess that is less confusing than silently ignoring a file that is there but not readable?

Reply:
Yup, you're right!